include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -89,41 +131,351 @@ class WhileOpInterface {
   T impl;
 };
 
-// Walks the control of this component, and appends source information for leaf
-// nodes. It also appends a position attribute that connects the source location
-// metadata to the corresponding control operation.
-WalkResult
-getCiderSourceLocationMetadata(calyx::ComponentOp component,
-                               SmallVectorImpl<Attribute> &sourceLocations);
+//===----------------------------------------------------------------------===//
+// Lowering state classes
+//===----------------------------------------------------------------------===//
 
-// Tries to match a constant value defined by op. If the match was
-// successful, returns true and binds the constant to 'value'. If unsuccessful,
-// the value is unmodified.
-bool matchConstantOp(Operation *op, APInt &value);
+// An interface for the Calyx component lowering state. The `Loop` type must
```

Comment:
Maybe a comment on what the 'Calyx component lowering state' intends to do/solve.